lib/site_setting_extension.rb

```diff
@@ -344,6 +367,13 @@ def all_settings(
           true
         end
       end
+      .select do |setting_name, _|
+        if only_upcoming_changes
+          upcoming_change_metadata.key?(setting_name)
+        else
+          true
+        end
```

Comment:
```suggestion
        !only_upcoming_changes || upcoming_change_metadata.key?(setting_name)
```

Reply:
Would prefer to keep this explicit like the other ones above e.g. filter_area filter_categories